src/NServiceBus.Core.Tests/Msmq/MsmqExtensionsTests.cs

```diff
@@ -33,18 +32,22 @@ public void Teardown()
         }
 
         [Test]
-        public void GetPermissions_returns_queue_access_rights()
+        [TestCase(AccessControlEntryType.Allow)]
+        [TestCase(AccessControlEntryType.Deny)]
+        public void GetPermissions_returns_queue_access_rights(AccessControlEntryType providedAccessType)
         {
-            queue.SetPermissions(LocalEveryoneGroupName, MessageQueueAccessRights.WriteMessage | MessageQueueAccessRights.ReceiveMessage, AccessControlEntryType.Allow);
+            queue.SetPermissions(LocalEveryoneGroupName, MessageQueueAccessRights.WriteMessage | MessageQueueAccessRights.ReceiveMessage, providedAccessType);
             MessageQueueAccessRights? rights;
-            if (!queue.TryGetPermissions(LocalEveryoneGroupName, out rights))
+            AccessControlEntryType? accessType;
+            if (!queue.TryGetPermissions(LocalEveryoneGroupName, out rights, out accessType))
             {
                 Assert.Fail("Unable to read permissions off a queue");
```

Comment:
@andreasohlund - I can do that, just FYI, I did not make changes to that code. I do agree having the queue name will make it more easier as well.